docs/infrastructure.md

```diff
@@ -0,0 +1,139 @@
+
+#### Azure Bastion
+
+[Azure Bastion](https://learn.microsoft.com/en-us/azure/bastion/bastion-overview) provides secure RDP and SSH connectivity to all of the VMs in the virtual network for which it's provisioned.
+Using Azure Bastion protects your virtual machines from exposing RDP/SSH ports to the outside world, while still providing secure access using RDP/SSH.
+
+![Architecture](https://learn.microsoft.com/en-us/azure/bastion/media/bastion-overview/architecture.png)
+
+### Cloud Infrastructure
+
+#### Resources
+
+Add the following resources:
+
+| Resource name           | Description                                                                                                                                                                                                                       |
+|-------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `Key Vault`             | [Azure Key Vault](https://learn.microsoft.com/en-us/azure/key-vault/general/overview) is a cloud service for securely storing and accessing secrets.                                                                              |
+| `App Configuration`     | [Azure App Configuration](https://learn.microsoft.com/en-us/azure/azure-app-configuration/overview) provides a service to centrally manage application settings and feature flags                                                                             |
+| `Storage Account`       | Storage account configured as [Data Lake](https://learn.microsoft.com/en-us/azure/storage/blobs/data-lake-storage-introduction) with hierachical namespace to host schema registry.                                               |
+| `Container Registry`    | [Azure Container Registry](https://learn.microsoft.com/en-us/azure/container-registry/container-registry-intro) is a managed registry service based on the open-source Docker Registry 2.0.                                       |
+| `Azure Monitor`         | [Azure Monitor](https://learn.microsoft.com/en-us/azure/azure-monitor/overview) is a comprehensive monitoring solution for collecting, analyzing, and responding to monitoring data from your cloud and on-premises environments. |
+| `Azure Managed Grafana` | [Azure Managed Grafana](https://learn.microsoft.com/en-us/azure/managed-grafana/overview) is a data visualization platform built on top of the Grafana software by Grafana Labs.                                                  |
+
+#### Private Endpoints
+
+![Private Endpoints](./assets/infrastructure-endpoints.png)
+
+`Key Vault`, `App Configuration`, `Storage Account`, and `Container Registry` are configured with private endpoints to restrict access to these resources within the Virtual Network (VNet).
```

Comment:
maybe we should mention that the private endpoint for storage account prevents to use the schema registry in dataflows

Reply:
Yes, added a note on this point